app/auth/auth_service.ts

```diff
@@ -50,11 +50,26 @@ export class AuthService {
         } else {
           this.onUserRpcError(error);
         }
+      })
+      .finally(() => {
+        setTimeout(() => {
+          this.startRefreshTimer();
+        }, 1000); // Wait a second before starting the refresh timer so we can grab the session duration.
```

Comment:
not sure I understand why the 1s delay is needed, shouldn't the cookie be set on the response (once then() is done?)

Reply:
In my testing about half of the time the cookie value was set by the time this ran, and the other half of the time it wasn't.

Not sure exactly how/when cookies are updated from an XHR. It's possible I just need to wait a tick instead of a whole second - but this is a longer running refresh loop that will happen in minutes to hours, so it doesn't matter much.